src/data/seq/seq.lean

```diff
@@ -41,6 +41,17 @@ def cons (a : α) : seq α → seq α
 /-- Get the nth element of a sequence (if it exists) -/
 def nth : seq α → ℕ → option α := subtype.val
 
+/-- A sequence terminates at position `n` if the value at position `n` equals `none`. -/
+def terminates_at (s : seq α) (n : ℕ) : Prop := s.nth n = none
```

Comment:
Should this be called `terminates_by`?

Reply:
Exactly, `terminates_at` sounds too much like that is the exact place it terminates.

Reply:
Fixed.